packages/menu-surface/index.tsx

```diff
@@ -0,0 +1,357 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import * as React from 'react';
+import * as classnames from 'classnames';
+import {MDCMenuSurfaceFoundation, Corner} from '@material/menu-surface';
+
+export interface MenuSurfaceProps {
+  className: string,
+  anchorElement: HTMLElement,
+  anchorCorner: number,
+  anchorMargin: object,
+  styles: object,
+  coordinates: {
+    x?: number,
+    y?: number
+  },
+  onClose: () => void,
+  onOpen: () => void,
+  onKeyDown: (event: React.KeyboardEvent) => void,
+  quickOpen: boolean,
+  open: boolean,
+  fixed: boolean
+};
+
+type MenuSurfaceState = {
+  transformOrigin: string,
+  maxHeight?: number,
+  styleLeft?: number,
+  styleRight?: number,
+  styleTop?: number,
+  styleBottom?: number,
+  classList: Set<string>
+};
+
+class MenuSurface extends React.Component<MenuSurfaceProps & React.HTMLProps<HTMLDivElement>, MenuSurfaceState> {
+  menuSurfaceElement_: React.RefObject<HTMLDivElement> = React.createRef();
+  previousFocus_: HTMLElement | null = null;
+  foundation_: MDCMenuSurfaceFoundation;
+  handleWindowClick_: EventListener;
+  registerWindowClickListener_: () => void;
+  deregisterWindowClickListener_: () => void;
+  firstFocusableElement_: HTMLElement | null;
+  lastFocusableElement_: HTMLElement | null;
+
+  state = {
+    transformOrigin: '',
+    maxHeight: undefined,
+    styleLeft: undefined,
+    styleRight: undefined,
+    styleTop: undefined,
+    styleBottom: undefined,
+    classList: new Set(),
+  } as MenuSurfaceState;
+
+  static defaultProps = {
+    className: '',
+    styles: {},
+    anchorElement: undefined,
```

Comment:
I'd probably avoid adding keys that have `undefined` values. I'm not sure if it affects how TS 3.0's defaultProps handling works, but it feels a bit odd to me. Just leaving the keys out is probably preferred